### DISCUSSION TITLE
Help with using ImageMagick for Montage + Command Line (updated question in comment)

### DISCUSSION THREAD

**Discussion by ezrabutler** [ImageMagick/ImageMagick#3022] [Digital Image Processing] [answered]
Hi,

I'm trying to 1) create a montage of an array of images, 2) insert it into top of a specific sized canvas (24"x40") (resizing the montage if necessary, but not altering the size of the canvas) , 3) and create a footer to be printed at the bottom of the canvas.

I need to do this on Node.js, so I am using GM (with ImageMagick).
1. I have easily: 
```

g = gm()
list.forEach(function(p){
    g.montage(p);
})
g.geometry('+81+81')
g.density(5000,5000)
.write('./images/montage.jpg', function(err) {
    if(!err) console.log("Written montage image.");
});>
```

2. While I'd rather do this all in one function, when I write
```
gm(2400, 4000, 'white')
.append('./images/montage.jpg',false)
.write('./images/blank-with-montage.jpg', function(err) {
    if(!err) console.log("Written image.");
});

```
it 1) modifies the size of the canvas, and 2) it puts the the montage image at the bottom of the image as opposed to the top, and adding `.gravity("North")` doesn't seem to do anything to help. (when I do `.composite(image)` it actually places the blank sheet in the middle of the montage.

3. I'm assuming I would, once this is done, I shouldn't be using `.title`, I should be using `.drawText` (but it is not extremely clear how to translate the command line to the node order.
This is what the montage function comes out looking like
![montage-test](https://user-images.githubusercontent.com/226214/102935984-d7a4f180-446c-11eb-9495-2002f6a70277.jpg)
This is basically how I want the page to be structured. 
But since I am sending it out to print, I need to maintain the correct file size.
<img width="432" alt="Screen Shot 2020-12-21 at 5 51 40 PM" src="https://user-images.githubusercontent.com/226214/102936027-f4d9c000-446c-11eb-96ef-e5f0f85655b1.png">

I appreciate any help/push in the correct direction.

**Comment by ezrabutler** [comment]
Even trying to do this in command line is causing me problems:
I can create a montage, and I can resize that to fit into the file needed, but I'm not able to put that file on the white background in the correct place.

```
magick montage -density 600 Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg \
          -geometry +81+81  miff:- | convert miff:- -resize 2300x4000  resized-montage.jpg
```
I can also create the background 
```
magick convert -size 2400x4000 xc:white blank.jpg 
```
I can't figure out how to create a file with the results of the first command on top of the second command.

**Comment by snibgo** [comment]
If "on top" means you want to composite one image over the other:
```
magick blank.jpg resized-montage.jpg -composite out.jpg
```
However, I would never use JPG for intermediate files, because they are lossy and only 8 bits/channel.

**Reply by ezrabutler** [reply]
Thank you!
So I've gotten this code working:
```
convert -size 2400x4000 canvas:white \( montage-v2.jpeg -resize 2300x4000 \) -gravity north -geometry +0+200 -composite new-canvas.jpg
```

But would love to not have to have montage-v2.jpeg already created, rather, to include this code instead, so as to forego any intermediate channel:

```
montage -density 600 Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg \
          -geometry +81+81 miff:- | convert miff:- 
```

Or must I do multiple steps?

Thank you

**Comment by snibgo** [comment, marked as answer]
You could do it like this. For clarity, I have broken it into lines.
```
magick montage -density 600 Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg Aqua.jpg Gold.jpg Brass.jpg \
Aqua.jpg \
          -geometry +81+81 \
 miff:- | magick \
-size 2400x4000 xc:white \
\( miff:- -resize 2300x4000  \) \
-gravity north -geometry +0+200 \
-composite \
 resized-montage.jpg
```

**Reply by ezrabutler** [reply]
Thank you so much! I now understand the point of the `miff:-`.
